### PR TITLE
Document: add guidance on dependency management

### PR DESCRIPTION
**Pull request by ahrtr**
cc @mitake @ptabor @serathius @spzala @jmhbnz @chaochn47 @fuweid @tjungblu

### CODE REVIEW

**Comment by ahrtr** [comment]
Currently the dependabot schedule interval is weekly, usually human intervention is required each time when dependabot automatically opens some PRs to bump dependencies.

I expect all contributors can take turn to take care of bumping the dependencies. For example, @jmhbnz will take care of it this week in https://github.com/etcd-io/etcd/pull/15737. Anyone else can drop a message under the PR to mention the willingness to do it next week.

**Review by fuweid — APPROVED** [review]
LGTM

**Comment by fuweid** [comment]
> Anyone else can drop a message under the PR to mention the willingness to do it next week.

Sounds good to me.

**Review by jmhbnz — COMMENTED** [review]
In general thos looks good - I will test tonight when I update this weeks pr and will suggest any changes if I run into any problems.

**Review by jmhbnz — APPROVED** [review]
LGTM - I followed the guide last night for #15737 and all worked well, thanks @ahrtr 🙏🏻

**Review by chaochn47 — APPROVED** [review]
LGTM. Thanks @ahrtr It's a big upstream ops win that the process is documented.

**Comment by chaochn47** [comment]
> > Anyone else can drop a message under the PR to mention the willingness to do it next week.
> 
> Sounds good to me.

I can give it a try after @fuweid's turn.

**Review by spzala — APPROVED** [review]
Very nice. Thanks @ahrtr

**Comment by ahrtr** [comment]
cc @pchan @ArkaSaha30

File: Documentation/contributor-guide/dependency_management.md [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+Please see guidance below.
+
+## Dependencies used in workflows
+The PRs which automatically bump dependencies (see examples below) used in workflows are fine, and can be approved & merged directly as long as all checks are successful.
+- [build(deps): bump github/codeql-action from 2.2.11 to 2.2.12](https://github.com/etcd-io/etcd/pull/15736)
+- [build(deps): bump actions/checkout from 3.5.0 to 3.5.2](https://github.com/etcd-io/etcd/pull/15735)
+- [build(deps): bump ossf/scorecard-action from 2.1.2 to 2.1.3](https://github.com/etcd-io/etcd/pull/15607)
+
+## Bumping order
+When multiple etcd modules depend on the same package, please bump the package version for all the modules in the correct order. The rule is simple:
```
Comment (pchan) [inline Documentation/contributor-guide/dependency_management.md:28]: Can we please add some context on how to identify these dependencies (for example how did we hit upon : `github.com/spf13/cobra`). These typically come from code changes so will something like the following be a good prefix to identify the dependency changes

> git clone https://github.com/etcd-io/etcd.git 
> go mod tidy  //check output
>

**Comment by pchan** [comment]
>I expect all contributors can take turn to take care of bumping the dependencies. For example, @jmhbnz will take care of it this week in https://github.com/etcd-io/etcd/pull/15737. Anyone else can drop a message under the PR to mention the willingness to do it next week.

I will take the one after @chaochn47, but we need a better way to manage this. Should we use google sheets to track who is doing for which week or make it incumbent on the person doing the activity to identify the next person.

File: Documentation/contributor-guide/dependency_management.md (hunk 1 above)
Comment (ahrtr) [reply Documentation/contributor-guide/dependency_management.md:28]: Please read the paragraph right below` # Main branch`

**Comment by ahrtr** [comment]
> Should we use google sheets to track who is doing for which week

We don't have much active contributors, I think dropping a message (see https://github.com/etcd-io/etcd/pull/15744#issuecomment-1515772835, pasted below) under the previous PR might be enough.  But I also created a google sheet https://docs.google.com/spreadsheets/d/1DDWzbcOx1p32MhyelaPZ_SfYtAD6xRsrtGRZ9QXPOyQ/edit#gid=0, anyone with the link have edit permission. 
> For example, @jmhbnz will take care of it this week in https://github.com/etcd-io/etcd/pull/15737. Anyone else can drop a message under the PR to mention the willingness to do it next week.

File: Documentation/contributor-guide/dependency_management.md (hunk 1 above)
Comment (pchan) [reply Documentation/contributor-guide/dependency_management.md:28]:
```suggestion
When multiple etcd modules depend on the same package, please close the dependabot PRs and bump the package version for all the modules in the correct order. The rule is simple:
```
I was looking at the earlier fix (https://github.com/etcd-io/etcd/pull/15313) where dependabot changes were being restricted to root and thought we won't be getting these PRs anymore. Is it incumbent on the person raising the request to close the dependabot PRs ?

Comment (ahrtr) [reply Documentation/contributor-guide/dependency_management.md:28]: yes, the related PRs opened by dependabot need to be closed. Updated the section "## Steps to bump a dependency". thx

**Comment by ahrtr** [comment]
Thanks for the feedback. 

@fuweid please take care of the PRs opened automatically opened by dependabot this week. thx

**Comment by sharathsivakumar** [comment]
@jmhbnz I would like to help with this.

**Comment by jmhbnz** [comment]
> @jmhbnz I would like to help with this.

Thanks for volunteering and adding your name to the roster @sharathsivakumar! 🙏🏻

**Pull request merged**